doc/src/BCMPackage.rst

```diff
@@ -18,10 +23,17 @@ bcm_package
 
 This setups a non-boost package in cmake. This is similar to ``bcm_boost_package`` but with extra flexibility as it does not have boost specific conventions. This function creates a library target that will be installed and exported. In addition, the target will be setup to auto-link for the tests.
 
+.. "setup to auto-link for the tests". This is a bit unclear. Maybe say
+   "Subsequent uses of 'bcm_test' will automatically include this target as a
+   dependency" or something along these lines.
+
 .. option:: <package-name>
 
 The name of the package. This is both the name of the library target and the cmake package config that can be used with ``find_package(<package-name>)``.
 
+.. This is inconsistent with 'bcm_boost_package' below. Why does Boost have to
+   be special?
```

Comment:
The `bcm_boost_package` does it differently to enforce that a boost library's name starts with `boost_`, the `find_package` name starts with `boost_`, and the imported target starts with `boost::`. Alternatively, we could check that the name starts with `boost_`, if we want to enforce the naming. This could makes things more consistent and clearer.